lang/bash/.transpile

```diff
@@ -62,22 +68,22 @@ transpile_for(){
   if [[ "$code" =~ [a-zA-Z_0-9],[a-zA-Z_0-9] ]]; then
     local key="$(echo "$code" | awk '{ print $2 }' | awk -F',' '{ print $1 }')"
     local value="$(echo "$code" | awk '{ print $2 }' | awk -F',' '{ print $2 }')"
-    code="$code\n$indent$value=\"\${$arr[\$$key]}\""
+    code="$code\n$indent$value=$(get_array_code $arr "\$$key")"
     code="${code/,$value/}"
     code="${code/ of / in }"
-    code="${code/ $arr/ \"\${!$arr\[@\]\}\"}";
-  else
+    code="${code/ $arr/ $(get_array_code "!$arr" @)}";
+  elif [[ ! "$arr" == "\$@" ]]; then
     # iterate over indexed array
     local key="$(echo "$code" | awk '{ print $2 }')"
-    [[ ! "$arr" == "\$@" ]] && code="${code/ $arr/ \"\${$arr\[@\]\}\"}"
+    code="${code/ $arr/ $(get_array_code $arr @)}"
   fi
   echo -e "$code" | transpile_all
 }
 
 transpile_if(){
   push stack "fi"
-  code="${1/if not/if \!}"
-  firstvar="${code/\! /}"
+  code="${1/if not/if !}"
+  firstvar="${code/! /}"
```

Comment:
Better yet, double-quoting the `!` seems to fix it as well.
I don't have a clue why travis is doing this, I googled for hours and haven't been able to find anyone else with the same problem. It's very weird.